### ISSUE TITLE
QoS - API and implementation for Liveliness and Deadline event callbacks

### ISSUE THREAD

**Issue by minggangw** [RobotWebTools/rclnodejs#492]

**Comment by minggangw** [comment]
Continued from #498 

>for this thing I need to add one more paramter in the methods which is used for creating Publisher/Subsciber, so I m asking if need to embed this new parameter inside options object or add just add a new paramter field.
What guidelines is followed I dont know

We suggest aligning with interfaces of rclpy, there is no guideline for designing an interface

>what is the use of timerCallback(),I have seen that at the end this function calls rcl_timer.c but cant understand what is its role

`timer.callback()` will trigger the callback function passed when creating this timer

**Comment by caelinsutch** [comment]
Is this taken?

**Comment by minggangw** [comment]
No, it' not. Please feel free to take it if you are interested in.

**Comment by minggangw** [comment]
Fixed by #1159
